### DISCUSSION TITLE
Output in PSObject instance

### DISCUSSION THREAD

**Discussion by sharpninja** [chocolatey/choco#2294] [Ideas]
I would like to add a feature where if the user adds -AsPSObject as a parameter that the output would be in a PSObject to aid in automation.  Would such a PR be welcome?

**Comment by pauby** [comment]
What benefit would that provide over outputting information in a CSV format (as we can do now with some commands) and converting it using `Convert-FromCSV`?

**Reply by sharpninja** [reply]
The project is a generic wrapper around any repository: WinGet, Choco and Scoop built in and extensible for future repos.  Having PSObject output with consistent properties would allow aggregating results more easily.

**Comment by TheCakeIsNaOH** [comment]
This might work for what you want to do:
https://github.com/ethanbergstrom/Foil

It is a PowerShell Crescendo wrapper for choco.
